### DISCUSSION TITLE
export plugins in language-service alone

### DISCUSSION THREAD

**Discussion by litingyes** [vuejs/language-tools#4636] [Ideas]
I want to simply expand the script/style language in the vue SFC, but it is not very convenient to reuse it at present, because only a full function including template is exported, and it is expected to be freely combined

> https://github.com/vuejs/language-tools/tree/master/packages/language-service/lib/plugins

**Comment by johnsoncodehk** [comment]
Does the method of filtering specific plugins work for your use case?

```ts
const wrantedPlugins = new Set([
	'css',
	'html',
	'json',
	// ...
]);
const plugins = getFullLanguageServicePlugins(ts).filter(plugin => wrantedPlugins.has(plugin.name!));
```

**Reply by litingyes** [reply]
Yes, it seems to meet my current needs.
